kernel/src/schema/mod.rs

```diff
@@ -321,6 +399,20 @@ impl StructType {
         let _ = get_leaves.transform_struct(self);
         (get_leaves.names, get_leaves.types).into()
     }
+
+    /// Applies physical name mappings to this field. If the `column_mapping_mode` is
+    /// [`ColumnMappingMode::Id`], then each StructField will have its parquet field id in the
+    /// [`ColumnMetadataKey::ParquetFieldId`] metadata field.
+    ///
+    /// NOTE: Caller affirms that the schema was already validated by
+    /// [`crate::table_features::validate_schema_column_mapping`], to ensure that annotations are
+    /// always and only present when column mapping mode is enabled.
+    pub fn make_physical(&self, column_mapping_mode: ColumnMappingMode) -> Self {
```

Comment:
interesting we didn't have this before?

Reply:
Yeah it's a consequence of how we build the physical schema. StateInfo holds the list of transformed StructFields, and then we put it all into a struct at the end. This might indicate that StateInfo isn't the right abstraction, but we can handle this later

Reply:
maybe don't make pub?